test/lsp.py

```diff
@@ -1408,6 +1408,38 @@ def test_custom_includes(self, solc: JsonRpcProcess) -> None:
         self.expect_equal(len(diagnostics), 1, "no diagnostics")
         self.expect_diagnostic(diagnostics[0], code=2018, lineNo=5, startEndColumns=(4, 62))
 
+    def test_remapping(self, solc: JsonRpcProcess) -> None:
+        self.setup_lsp(solc, expose_project_root=False)
+        solc.send_notification(
+            'workspace/didChangeConfiguration', {
+                'settings': {
+                    'remappings': [
+                        {
+                            'context': '',
+                            'prefix': '@other/',
+                            'target': f"{self.project_root_dir}/other-include-dir/otherlib/"
+                        }
+                    ]
+                }
+            }
+        )
+        published_diagnostics = self.open_file_and_wait_for_diagnostics(solc, 'include-paths/remapped-include')
```

Comment:
It looks to me like you need the content of this file only in this function and not for any of the "in file defined" test mechanics
Given that it serves no purpose other than in this function here, it would be nicer to just embed file test content directly like it is done in the function `test_textDocument_opening_two_new_files_edit_and_close`

You can still use the `otherlib.sol` for the import of course.

Reply:
👍